packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -535,20 +535,88 @@ String _generateDelegateClass({
 }
 
 class LocalizationsGenerator {
+  /// Initializes [inputDirectory], [outputDirectory], [templateArbFile],
+  /// [outputFile] and [className].
+  ///
+  /// Throws an [L10nException] when a provided configuration is not allowed
+  /// by [LocalizationsGenerator].
+  ///
+  /// Throws a [FileSystemException] when a file operation necessary for setting
+  /// up the [LocalizationsGenerator] cannot be completed.
+  factory LocalizationsGenerator({
+    @required FileSystem fileSystem,
+    @required String inputPathString,
+    String outputPathString,
+    @required String templateArbFileName,
+    String outputFileString,
+    @required String classNameString,
+    List<String> preferredSupportedLocales,
+    String headerString,
+    String headerFile,
+    bool useDeferredLoading = false,
+    String inputsAndOutputsListPath,
+    bool useSyntheticPackage = true,
+    String projectPathString,
+    bool areResourceAttributesRequired = false,
+    String untranslatedMessagesFile,
+    bool usesNullableGetter = true,
+  }) {
+    final Directory projectDirectory = projectDirFromPath(fileSystem, projectPathString);
+    final Directory inputDirectory = inputDirectoryFromPath(fileSystem, inputPathString, projectDirectory);
+    final Directory outputDirectory = outputDirectoryFromPath(fileSystem, outputPathString ?? inputPathString, useSyntheticPackage, projectDirectory);
+    return LocalizationsGenerator._(
+      fileSystem,
+      useSyntheticPackage: useSyntheticPackage,
+      usesNullableGetter: usesNullableGetter,
+      className: classNameFromString(classNameString),
+      projectDirectory: projectDirectory,
+      inputDirectory: inputDirectory,
+      outputDirectory: outputDirectory,
+      templateArbFile: templateArbFileFromFileName(templateArbFileName, inputDirectory),
+      baseOutputFile: outputDirectory.childFile(outputFileString),
+      preferredSupportedLocales: preferredSupportedLocalesFromLocales(preferredSupportedLocales),
+      header: headerFromFile(headerString, headerFile, inputDirectory),
+      useDeferredLoading: useDeferredLoading ?? false,
```

Comment:
I replaced `_setUseDeferredLoading` with just a default of `false` 
https://github.com/flutter/flutter/blob/76bc462520ba462f7bfe810db0f2f1a288f68e97/packages/flutter_tools/lib/src/localizations/gen_l10n.dart#L901-L906
since that seemed to be intention of the `= false` in `initialize`
https://github.com/flutter/flutter/blob/76bc462520ba462f7bfe810db0f2f1a288f68e97/packages/flutter_tools/lib/src/localizations/gen_l10n.dart#L690